File: framework/languages/java/samples/java/GameQuest/Server/GameApi/src/main/java/systems/zlink/samples/gamequest/server/gameapi/handlers/CompleteMissionHandler.java
package systems.zlink.samples.gamequest.server.gameapi.handlers;

import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;
import systems.zlink.samples.gamequest.server.gameapi.application.GameplayActionService;
import systems.zlink.samples.gamequest.shared.contracts.Messages;

@ZLinkHandlerGroup("gameapi")
public final class CompleteMissionHandler
    implements ZLinkRequestHandler<Messages.CompleteMissionReq, Messages.CompleteMissionRes> {
    private final GameplayActionService actions;

    public CompleteMissionHandler(GameplayActionService actions) {
        this.actions = actions;
    }

    @Override
    public Messages.CompleteMissionRes handle(Messages.CompleteMissionReq request, ZLinkRequestContext context) {
        return actions.completeMission(request);
    }
}

File: framework/languages/java/samples/kotlin/GameQuest/Server/GameApi/src/main/kotlin/systems/zlink/samples/kotlin/gamequest/server/gameapi/handlers/CompleteMissionHandler.kt
package systems.zlink.samples.kotlin.gamequest.server.gameapi.handlers

import systems.zlink.framework.channels.ZLinkRequestContext
import systems.zlink.framework.kotlin.ZLinkSuspendingRequestHandler
import systems.zlink.framework.handlers.ZLinkHandlerGroup
import systems.zlink.samples.kotlin.gamequest.server.gameapi.application.GameplayActionService
import systems.zlink.samples.kotlin.gamequest.shared.contracts.CompleteMissionReq
import systems.zlink.samples.kotlin.gamequest.shared.contracts.CompleteMissionRes

@ZLinkHandlerGroup("gameapi")
class CompleteMissionHandler(
    private val actions: GameplayActionService,
) : ZLinkSuspendingRequestHandler<CompleteMissionReq, CompleteMissionRes> {
    override suspend fun handle(request: CompleteMissionReq, context: ZLinkRequestContext): CompleteMissionRes =
        actions.completeMission(request)
}
